OSDOCS-9244: Machine API updates for Nutanix failure domain support

jeana-redhat opened the pull request:
Version(s):
4.15+

Issue:
[OSDOCS-9244](https://issues.redhat.com//browse/OSDOCS-9244)

Link to docs preview:
* [Modifying a compute machine set by using the CLI](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/modifying-machineset#machineset-modifying_modifying-machineset) (small tweaks)
* [Failure domains for Nutanix clusters](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-nutanix#mapi-failure-domain-nutanix_creating-machineset-nutanix) (compute)
* [Failure domains for Nutanix clusters](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-configuration#mapi-failure-domain-nutanix_cpmso-configuration) (control plane)
* [Failure domain platform support and configuration](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-resiliency#cpmso-failure-domains-provider_cpmso-resiliency)
* [Distributing compute machines across failure domains](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/post_installation_configuration/adding-nutanix-failure-domains.html#post-installation-adding-nutanix-failure-domains-compute-machines_adding-failure-domains-to-an-existing-nutanix-cluster)

QE review:
- [x] QE has approved this change.

Additional information:

jeana-redhat commented on modules/mapi-failure-domain-nutanix.adoc:
```diff
@@ -0,0 +1,19 @@
+To add or update the failure domain configuration on a Nutanix cluster, you must make coordinated changes to several resources.
+The following actions are required:
+
+. Modify the cluster infrastructure custom resource definition (CRD).
```
```suggestion
. Modify the cluster infrastructure custom resource (CR).
```

yannickstruyf3 commented:
LGTM

jeana-redhat commented:
@sgaoshang @sunzhaohua2 this PR has some machine set content and some specific failure domain content. Could you please take a look from a QA perspective?

tmalove commented:
/remove-label peer-review-needed
/label peer-review-in-progress

tmalove commented on modules/post-installation-adding-nutanix-failure-domains-compute-machines-edit.adoc:
```diff
@@ -75,14 +82,27 @@ spec:
 ----
-. Note the value of `spec.replicas`, as you need it when scaling the machine set to apply the changes.
+
+. Note the value of `spec.replicas`, as you need it when scaling the compute machine set to apply the changes.
```
```suggestion
. Note the value of `spec.replicas`, because you need it when scaling the compute machine set to apply the changes.
```

tmalove commented:
/remove-label peer-review-in-progress
/label peer-review-done
@jeana-redhat just one nit/suggestion, otherwise it looks good.

jeana-redhat commented:
refreshing PR

huali9 commented on modules/post-installation-adding-nutanix-failure-domains-compute-machines-replace.adoc:
```diff
@@ -0,0 +1,223 @@
+<3> The values in the `<providerSpec>` section of the compute machine set CR are platform-specific. For more information about `<providerSpec>` parameters in the CR, see the sample compute machine set CR configuration for your provider.
+--
+
+. Configure the new compute machine set to use the first failure domain by adding the following to the `spec.template.spec.providerSpec.value` stanza in the `<new_machine_set_name_1>.yaml` file.
```
maybe change `adding` to `updating or adding` should be better, because if using the 1st method "** Copy an existing compute machine set configuration into a new file by running the following command", then these fields already exist , we just need to modify their values. 
And if using the 2nd method "** Create a blank YAML file named `<new_machine_set_name_1>.yaml` with your preferred text editor and include the required values for your new compute machine set.", these fields don't exist, then we should add.

jeana-redhat replied:
Thanks, I like the added accuracy :nerd_face:

huali9 commented on modules/post-installation-adding-nutanix-failure-domains-compute-machines-replace.adoc:
```diff
@@ -0,0 +1,223 @@
+<new_machine_set_name_2>   1         1         1       1           4m12s
+----
+
+* To verify that the compute machines without the updated configuration are deleted, list the machines that are managed by the updated compute machine set by running the following command:
```
maybe change `list the machines that are managed by the updated compute machine set by running the following command:` to `list the machines by running the following command:`, because in the example later we list the machines for both the original  compute machine set and updated compute machine set

huali9 commented on modules/post-installation-adding-nutanix-failure-domains-compute-machines-replace.adoc:
```diff
@@ -0,0 +1,223 @@
++
+[source,terminal]
+----
+$ oc get -n openshift-machine-api machines -l machine.openshift.io/cluster-api-machineset=<machine_set_name>
```
maybe should not specify `-l machine.openshift.io/cluster-api-machineset=<machine_set_name>`, because it can only list the machines of this machine set, but in the example later it lists machines for two machine set.

jeana-redhat commented on modules/post-installation-adding-nutanix-failure-domains-compute-machines-edit.adoc:
```diff
@@ -75,14 +82,27 @@ spec:
+----
+NAME                        PHASE     TYPE         REGION      ZONE         AGE
+<machine_name_original_1>   Running   m6i.xlarge   us-west-1   us-west-1a   4h
+<machine_name_original_2>   Running   m6i.xlarge   us-west-1   us-west-1a   4h
```
Perfect, thank you! A lot of the sample output I have is AWS, but very glad to use Nutanix instead :)

jeana-redhat commented:
@huali9 great feedback, thank you! I incorporated all of your fixes. I also added statements to [Modifying a compute machine set by using the CLI](https://70056--ocpdocs-pr.netlify.app/openshift-enterprise/latest/machine_management/modifying-machineset#machineset-modifying_modifying-machineset) to explain that the example output throughout shows AWS values.

huali9 commented:
Thank you @jeana-redhat
